pkg/controller/migmigration/task.go

```diff
@@ -583,14 +625,20 @@ func (t *Task) next() {
 		break
 	}
 	if current == -1 {
-		if t.canceled() {
-			t.Phase = CancelItinerary[0].phase
+		if t.failed() || t.canceled() {
+			current = 0
 		} else {
-			t.Phase = Completed
+			current = len(t.Itinerary) - 1
 		}
-		return
+		t.Phase = t.Itinerary[current].phase
 	}
-	for n := current + 1; n < len(t.Itinerary); n++ {
+
+	return current
+}
+
+// Advance the task to the next phase.
+func (t *Task) next(currentPhaseIndex int) {
```

Comment:
Please explain how this change is related to and required for this feature.  Else, revert.